### ISSUE TITLE
SearchableMixin wont work like this with more than one class

### ISSUE THREAD

**Issue by coolkau** [miguelgrinberg/microblog#107]
I tried to use the events for my own purposes and I came across the following issue with the after_commit method:

@classmethod
def after_commit(cls, session):
    [...]
    session._changes = None

The event is triggered and hits the first after_commit method. This first after_commit will reset the session changes. So if you have have a second class which is listening on the after_commit and queries the changes are wiped and run into an error because _changes is None at that point. At least that happened in my version. 

Other than that thanks for the great tutorial. I'd love to see a bit more about testing of the routes.

**Comment by miguelgrinberg** [comment]
I did not consider this case, you are correct. the `session._changes` attribute needs to independently store changes for multiple classes. I will fix it and update the tutorial.

**Comment by coolkau** [comment]
One more thing I came across which is worth mentioning during a rewrite is: You rely on ` session.deleted` to be populated when a delete happens. However, you can delete entities without the ` session.deleted` being filled (both before and after commit). (No clue if it is a bug with sqlAlchemy or not)

This one will leave the ` session.deleted` empty, yet deletes the entity; meaning  -- in case of the tutorial -- the search index is not deleted.
`Model.query.filter_by(...).delete()`

This one will fill the ` session.deleted` 
```
Entity = Model.query.filter_by(...).one()
db.session.delete(Entity)
```

So one should at least be aware of this so that one can write really good test making sure one does not anything stupid...

**Comment by miguelgrinberg** [comment]
@coolkau Yes, the solution that I provide integrates the full-text search with the SQLAlchemy ORM, as shown throughout the tutorial. Any database operations that do not go through the scoped session will not work for this solution.

**Comment by coolkau** [comment]
@miguelgrinberg IMHO, it would be worth writing a comment about exactly that in the tutorial, as I dont really know the difference (until that issue). I dont think there is actually a single delete in the tutorial. So hobby coders like me will go and find the next best solution on stackoverflow, and wonder why it is not working much later... so it would be nice to have a warning... but up to you, its your tutorial after all.
